maven-plugin/src/main/java/org/openmrs/maven/plugins/GenerateDistro.java

```diff
@@ -0,0 +1,96 @@
+        options.add(O3_DISTRIBUTION);
+        String choice = wizard.promptForMissingValueWithOptions(GENERATE_DISTRO_PROMPT, null, null, options);
+
+        DistroProperties distroProperties;
+
+        Server server = new Server.ServerBuilder().build();
+
+        try {
+            server.setServerDirectory(Files.createTempDirectory("distro").toFile());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        File outputFile = new File(outputLocation);
+        switch (choice) {
+            case O2_DISTRIBUTION:
+                wizard.promptForRefAppVersionIfMissing(server, versionsHelper, GET_VERSION_PROMPT);
+                if (DistroHelper.isRefapp2_3_1orLower(server.getDistroArtifactId(), server.getVersion())) {
+                    distroProperties = new DistroProperties(server.getVersion());
+                } else {
+                    distroProperties = distroHelper.downloadDistroProperties(outputFile, server);
+                }
+                distroProperties.saveTo(outputFile);
+                break;
+
+            case O3_DISTRIBUTION:
+                wizard.promptForO3RefAppVersionIfMissing(server, versionsHelper, GET_VERSION_PROMPT);
+                Artifact artifact = new Artifact(server.getDistroArtifactId(), server.getVersion(),
+                        server.getDistroGroupId(), "zip");
+                Properties frontendProperties;
+                if (new Version(server.getVersion()).higher(new Version("3.0.0-beta.16"))) {
+                    frontendProperties = distroHelper.getFrontendProperties(server);
+                }
+                else {
+                    frontendProperties = PropertiesUtils.getFrontendPropertiesFromSpaConfigUrl(
+                            "https://raw.githubusercontent.com/openmrs/openmrs-distro-referenceapplication/"+ server.getVersion() +"/frontend/spa-build-config.json");
+                }
+                Properties configurationProperties = PropertiesUtils.getConfigurationProperty(artifact);
+                File file = distroHelper.downloadDistro(server.getServerDirectory(), artifact);
+                Properties backendProperties = PropertiesUtils.getDistroProperties(file);
+                Properties spaModuleProperty = PropertiesUtils.getModuleProperty("https://raw.githubusercontent.com/openmrs/openmrs-module-spa/master/pom.xml");
+                Properties allProperties = new Properties();
+                allProperties.putAll(backendProperties);
+                allProperties.putAll(spaModuleProperty);
+                allProperties.putAll(frontendProperties);
+                allProperties.putAll(configurationProperties);
+                distroProperties = new DistroProperties(allProperties);
+                distroProperties.saveTo(outputFile);
+                break;
+        }
+        logger.info(String.format("openmrs-distro.properties created successfully at %s", outputFile.getAbsolutePath()));
```

Comment:
I still see a `.` in the output file location.

Reply:
Strange, it is working fine for me.

```
wikum@msi:~/openmrs/distro$ mvn org.openmrs.maven.plugins:openmrs-sdk-maven-plugin:5.6.0-SNAPSHOT:generate-distro
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- openmrs-sdk-maven-plugin:5.6.0-SNAPSHOT:generate-distro (default-cli) @ standalone-pom ---

You can generate the distro.properties file for the following distributions:
1) 2.x Distribution
2) O3 Distribution

Which one do you choose? [1/2]: 2

You can generate the distro.properties file for the following versions:
1) Reference Application 3.0.0-SNAPSHOT
2) Reference Application 3.0.0-beta.16
3) Reference Application 3.0.0-beta.13
4) Reference Application 3.0.0-beta.12
5) Reference Application 3.0.0-beta.11
6) Other...

Which one do you choose? [1/2/3/4/5/6]: 1
[INFO] Configured Artifact: org.openmrs.distro:referenceapplication-frontend:3.0.0-SNAPSHOT:zip
[INFO] Copying referenceapplication-frontend-3.0.0-SNAPSHOT.zip to /tmp/distro9222204486787434693/frontend.zip
[INFO] Configured Artifact: org.openmrs.distro:referenceapplication-distro:3.0.0-SNAPSHOT:zip
[INFO] Copying referenceapplication-distro-3.0.0-SNAPSHOT.zip to /tmp/distro9222204486787434693/openmrs-distro.jar
[INFO] openmrs-distro.properties created successfully at /home/wikum/openmrs/distro/.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  14.671 s
[INFO] Finished at: 2024-02-19T15:33:56+05:30
[INFO] ------------------------------------------------------------------------
```

Reply:
Can we replace the `.` and the end with the distro properties file name?

Reply:
@dkayiwa I updated the pull request